### ISSUE TITLE
Document length-fitting on <select>

### ISSUE THREAD

**Issue by mangelozzi** [mdn/content#25073]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/max-width

### What specific section or headline is this issue about?

The Into and "Try it"

### What information was incorrect, unhelpful, or incomplete?

> The max-width [CSS](https://developer.mozilla.org/en-US/docs/Web/CSS) property sets the maximum width of an element. It prevents the [used value](https://developer.mozilla.org/en-US/docs/Web/CSS/used_value) of the [width](https://developer.mozilla.org/en-US/docs/Web/CSS/width) property from becoming larger than the value specified by max-width.

> max-width overrides [width](https://developer.mozilla.org/en-US/docs/Web/CSS/width), but [min-width](https://developer.mozilla.org/en-US/docs/Web/CSS/min-width) overrides max-width.

Although true a lot of the time, its not technically true all the time. The exceptions is what makes debugging code so hard. It might be worth adding a note for when this is not true, like with select inputs, `width` overrides `max-width`.

### What did you expect to see?

Information to help explain why `max-width` is not working on a `<select>` input.

### Do you have any supporting links, references, or citations?

https://stackoverflow.com/a/10795269/5506400

### Do you have anything more you want to share?

Hope one day form controls will be more consistent with their API. Every form control has some weird quirks one has to work around.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/max-width`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/max-width
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/max-width/index.md
* Last commit: https://github.com/mdn/content/commit/7a9f9baa25d9a7313bd6c62ef5ef585b28459c58
* Document last modified: 2023-02-21T06:01:00.000Z

</details>

**Comment by yarusome** [comment]
For the `select` element, what happens is not that `width` overrides `max-width` (at least for Gecko). When a `<select>` has its `width` larger than `max-width` in Gecko, the selection bar will *always* shrink to fit `max-width` just like a box with `overflow-x: hidden`, and the dropdown menu will *try* to shrink to fit `max-width`. But if the dropdown menu has an item longer than `max-width`, the dropdown menu will shrink to fit the length of the longest item instead, just like a box with `min-width: max-content` (and `min-width` overrides `max-width`).

---
A visual example:

Folded:
```
+-----------+-+
| The selec |v|
+-----------+-+
```
Expanded:
```
+-----------+-+
| The selec |v|
+-----------+-+--------------+
| A really, really long item |
| A short item               |
+----------------------------+ 
```
---
That being said, it would be nice to mention the difference of length-fitting methods between the selection bar and the dropdown menu, and possible non-interoperability between browsers in this regard.

**Comment by OnkarRuikar** [comment]
> That being said, it would be nice to mention the difference of length-fitting methods between the selection bar and the dropdown menu, and possible non-interoperability between browsers in this regard.

I don't think we should document browser specific behaviours and per element wise exceptional cases on the page.

> Although true a lot of the time, its not technically true all the time. It might be worth adding a note for when this is not true, like with select inputs, width overrides max-width

This is very open ended statement. Specific example demonstrating the case would be nice.

**Comment by estelle** [comment]
Suggestion:
1. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select
   - Add a` max-width` example on the `<select>` page.
   - Explanations of the issues of setting a `max-width` on a  `<select>` can be included in the `<select>` page.
   - If there are issues with selects and `min-width`, this would be the place to address those too.
2. https://developer.mozilla.org/en-US/docs/Web/CSS/max-width
- Briefly mention there is an issue with `max-width` on `<select>` . 
- Link from that sentence on the `max-width` page to the example on the `<select>` page. 

in other words, the select page is the appropriate venue for this issue to be addressed. We don't want to not include extra stuff on select in the`max-width` page other than providing a note that an issue exists with a link to where they can get more information about it.

**Comment by wbamberg** [comment]
> Explanations of the issues of setting a max-width on a <select> can be included in the <select> page.

Note we already have https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#styling_with_css , which could be a good home for this.

**Comment by OnkarRuikar** [comment]
> Information to help explain why `max-width` is not working on a `<select>` input.

It is not the issue with `max-width` or `<select>` element. It is about when `<fieldset>` is used as an input wraper. And in the StackOverflow [minimal reproducible](http://jsfiddle.net/e2H82/) if you use `<div>` instead of the fieldset it works as expected.

Fieldset is a special element with the following characteristic: [[ref](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset#styling_with_css)]
> The element has `min-inline-size: min-content` by default.

If you set `min-inline-size: 0` on fieldset it works as expected: https://jsfiddle.net/cp580nvk/5/

**Comment by estelle** [comment]
This issue is a `<fieldset>` issue rather than a `<select>` issue.

Closing this as the `<fieldset>` page already has a section dedicated to [styling with CSS](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/fieldset#styling_with_css).
